src/server/api/routes.rs

```diff
@@ -0,0 +1,275 @@
+/// Static routes should be registered first, followed by dynamic routes.
+///
+/// # Errors
+/// Errors if unable to register dynamic routes (e.g. if git repository cannot be opened)
+pub fn register_app<
+    T: Global + Clone + 'static,
+    U: MessageBody,
+    V: ServiceFactory<
+        ServiceRequest,
+        Response = ServiceResponse<U>,
+        Config = (),
+        InitError = (),
+        Error = Error,
+    >,
+>(
+    mut app: App<V>,
+    state: &T,
+) -> anyhow::Result<
+    App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<U>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+> {
+    app = app
+        .service(
+            web::scope("/_api").service(
+                web::scope("/versions")
+                    .service(
+                        web::resource(
+                            "/_publication/{publication}/_compare/{date}/{compare_date}/{path:.*}",
+                        )
+                        .to(versions),
+                    )
+                    .service(web::resource("/_publication/{publication}/_date/{date}").to(versions))
+                    .service(web::resource("/_publication/{publication}").to(versions))
+                    .service(web::resource("/_publication/{publication}/{path:.*}").to(versions))
+                    .service(
+                        web::resource("/_compare/{date}/{compare_date}/{path:.*}").to(versions),
+                    )
+                    .service(web::resource("/_date/{date}/{path:.*}").to(versions))
+                    .service(web::resource("/{path:.*}").to(versions)),
+            ),
+        )
+        .app_data(web::Data::new(state.clone()));
+
+    let initialized_app = register_dynamic_routes(app, state)?;
+    Ok(initialized_app)
+}
+
+/// Initialize all dynamic routes for the given Archive.
+///
+/// Dynamic routes are determined at runtime by looking at the stele's `dependencies.json` and `repositories.json` files
+/// in the authentication (e.g. law) repository.
+///
+/// # Errors
+/// Errors if unable to register dynamic routes (e.g. if git repository cannot be opened)
+fn register_dynamic_routes<T: MessageBody>(
+    mut app: App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<T>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+    state: &impl Global,
+) -> anyhow::Result<
+    App<
+        impl ServiceFactory<
+            ServiceRequest,
+            Response = ServiceResponse<T>,
+            Config = (),
+            InitError = (),
+            Error = Error,
+        >,
+    >,
+> {
+    let config = state.archive().get_config()?;
+    let stelae_guard = config
+        .headers
+        .and_then(|headers| headers.current_documents_guard);
+
+    if let Some(guard) = stelae_guard {
```

Comment:
Can the contents of the `if` and `else` blocks live in their own functions?

Reply:
Yes, moved the contents to two new functions.